compiler/src/iree/compiler/Codegen/Dialect/GPU/TargetUtils/ConfigUtils.cpp

```diff
@@ -215,14 +245,17 @@ LogicalResult setMatmulLoweringConfig(IREE::GPU::TargetAttr target,
   LDBG("Target Subgroup size: " << targetSubgroupSize);
   LDBG("Schedule: sizes [" << schedule->mSize << ", " << schedule->nSize << ", "
                            << schedule->kSize << "]");
-  LDBG("Schedule: tile counts [" << schedule->mTileCount << ", "
-                                 << schedule->nTileCount << ", "
-                                 << schedule->kTileCount << "]");
-  LDBG("Schedule: warp counts [" << schedule->mWarpCount << ", "
-                                 << schedule->nWarpCount << "]");
-
-  std::array<int64_t, 3> workgroupSize{
-      schedule->nWarpCount * targetSubgroupSize, schedule->mWarpCount, 1};
+  LDBG("Schedule: tile counts [" << schedule->mTileSizes << ", "
+                                 << schedule->nTileSizes << ", "
+                                 << schedule->kTileSizes << "]");
+  LDBG("Schedule: warp counts [" << schedule->mSubgroupCounts << ", "
+                                 << schedule->nSubgroupCounts << "]");
+
+  int64_t flatWorkgroupSize =
+      targetSubgroupSize *
+      ShapedType::getNumElements(schedule->nSubgroupCounts) *
+      ShapedType::getNumElements(schedule->mSubgroupCounts);
+  std::array<int64_t, 3> workgroupSize{flatWorkgroupSize, 1, 1};
```

Comment:
Interesting. Not anything to address but i'm curious how we are able to do this. Is this the new scf.forall tiling that makes it possible?

Reply:
The TileAndFuse pipeline only uses the flat workgroup size anyway, since it uses linearization and delinearization, so it doesn't really matter how the dimensions are split up.